dirty_cat/minhash_encoder.py

```diff
@@ -47,90 +51,86 @@ class MinHashEncoder(BaseEstimator, TransformerMixin):
         """
         return {"X_types": ["categorical"]}
 
-    def get_unique_ngrams(self, string, ngram_range):
-        """ Return the set of unique n-grams of a string.
-        Parameters
-        ----------
-        string : str
-            The string to split in n-grams.
-        ngram_range : tuple (min_n, max_n)
-        The lower and upper boundary of the range of n-values for different
-        n-grams to be extracted. All values of n such that min_n <= n <= max_n.
-        Returns
-        -------
-        set
-            The set of unique n-grams of the string.
+    def minhash(self, string: str, n_components: int,
+                ngram_range: Tuple[int, int]) -> np.array:
         """
-        spaces = ' '  # * (n // 2 + n % 2)
-        string = spaces + " ".join(string.lower().split()) + spaces
-        ngram_set = set()
-        for n in range(ngram_range[0], ngram_range[1] + 1):
-            string_list = [string[i:] for i in range(n)]
-            ngram_set |= set(zip(*string_list))
-        return ngram_set
-
-    def minhash(self, string, n_components, ngram_range):
-        """ Encode a string using murmur hashing function.
+        Encode a string using murmur hashing function.
+
         Parameters
         ----------
         string : str
             The string to encode.
         n_components : int
             The number of dimension of encoded string.
         ngram_range : tuple (min_n, max_n)
-        The lower and upper boundary of the range of n-values for different
-        n-grams to be extracted. All values of n such that min_n <= n <= max_n.
+            The lower and upper boundary of the range of n-values for different
+            n-grams to be extracted.
+            All values of n such that min_n <= n <= max_n.
+
         Returns
         -------
         array, shape (n_components, )
             The encoded string.
         """
         min_hashes = np.ones(n_components) * np.infty
-        grams = self.get_unique_ngrams(string, self.ngram_range)
+        grams = get_unique_ngrams(string, self.ngram_range)
         if len(grams) == 0:
-            grams = self.get_unique_ngrams(' Na ', self.ngram_range)
+            grams = get_unique_ngrams(' Na ', self.ngram_range)
         for gram in grams:
             hash_array = np.array([
                 murmurhash3_32(''.join(gram), seed=d, positive=True)
                 for d in range(n_components)])
             min_hashes = np.minimum(min_hashes, hash_array)
         return min_hashes / (2 ** 32 - 1)
 
-    def get_fast_hash(self, string):
+    def get_fast_hash(self, string: str) -> np.array:
         """
         Encode a string with fast hashing function.
         fast hashing supports both min_hash and minmax_hash encoding.
+
         Parameters
         ----------
         string : str
             The string to encode.
+
         Returns
         -------
-        array, shape (n_components, )
+        np.array of shape (n_components, )
```

Comment:
Is this consistent with numpy doc standard?

Long term, we should add a check of consistency, as scikit-learn has recently done.

Reply:
Indeed ! I'll leave it like that for now, but you're right that we need a consistency check. I've opened #305 to address that later on.

Reply:
I'd rather make sure that it renders well on the HTML rendering (which it might not if it is not valid numpy doc standard)